// @abp/ng.schematics/models/method.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const enums_1 = require("../enums");
const utils_1 = require("../utils");
class Method {
    constructor(options) {
        Object.assign(this, options);
    }
}
exports.Method = Method;
class Signature {
    constructor(options) {
        this.generics = '';
        this.modifier = enums_1.eMethodModifier.Public;
        this.parameters = [];
        this.returnType = '';
        Object.assign(this, options);
    }
}
exports.Signature = Signature;
class Body {
    constructor(options) {
        this.params = [];
        this.requestType = 'any';
        this.registerActionParameter = (param) => {
            const { bindingSourceId, descriptorName, name, nameOnMethod } = param;
            const camelName = utils_1.camel(name);
            const value = descriptorName ? `${descriptorName}.${camelName}` : nameOnMethod;
            switch (bindingSourceId) {
                case enums_1.eBindingSourceId.Model:
                case enums_1.eBindingSourceId.Query:
                    this.params.push(`${camelName}: ${value}`);
                    break;
                case enums_1.eBindingSourceId.Body:
                    this.body = value;
                    break;
                case enums_1.eBindingSourceId.Path:
                    const regex = new RegExp('{(' + camelName + '|' + name + ')}', 'g');
                    this.url = this.url.replace(regex, '${' + value + '}');
                    break;
                default:
                    break;
            }
        };
        Object.assign(this, options);
    }
}
exports.Body = Body;
