// @ngx-validate/core/esm5/lib/directives/validation-group.directive.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/directives/validation-group.directive.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { ChangeDetectorRef, Directive, Injector, Input, Optional, Self, SkipSelf, } from '@angular/core';
import { FormGroupDirective, FormGroupName } from '@angular/forms';
import { ReplaySubject, Subscription } from 'rxjs';
import { AbstractValidationDirective } from '../abstracts';
var ValidationGroupDirective = /** @class */ (function (_super) {
    tslib_1.__extends(ValidationGroupDirective, _super);
    function ValidationGroupDirective(injector, cdRef, groupName, groupRef, parentRef) {
        var _this = _super.call(this, injector) || this;
        _this.injector = injector;
        _this.cdRef = cdRef;
        _this.groupName = groupName;
        _this.groupRef = groupRef;
        _this.parentRef = parentRef;
        _this.status$ = new ReplaySubject(1);
        _this.submit$ = new ReplaySubject(1);
        _this.value$ = new ReplaySubject(1);
        _this.subs = new Subscription();
        return _this;
    }
    /**
     * @private
     * @return {?}
     */
    ValidationGroupDirective.prototype.subscribeToFormSubmit = /**
     * @private
     * @return {?}
     */
    function () {
        var _this = this;
        ((/** @type {?} */ (this.elRef.nativeElement))).onsubmit = (/**
         * @param {?} event
         * @return {?}
         */
        function (event) {
            if (_this.group.invalid)
                event.preventDefault();
            _this.submit$.next(_this.group);
            _this.cdRef.markForCheck();
        });
    };
    /**
     * @private
     * @return {?}
     */
    ValidationGroupDirective.prototype.subscribeToStatusChanges = /**
     * @private
     * @return {?}
     */
    function () {
        var _this = this;
        this.subs.add(this.group.statusChanges.subscribe((/**
         * @return {?}
         */
        function () {
            _this.status$.next(_this.group);
            _this.cdRef.markForCheck();
        })));
    };
    /**
     * @private
     * @return {?}
     */
    ValidationGroupDirective.prototype.subscribeToValueChanges = /**
     * @private
     * @return {?}
     */
    function () {
        var _this = this;
        this.subs.add(this.group.valueChanges.subscribe((/**
         * @return {?}
         */
        function () {
            _this.value$.next(_this.group);
            _this.cdRef.markForCheck();
        })));
    };
    /**
     * @return {?}
     */
    ValidationGroupDirective.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        if (!this.parentRef)
            this.subscribeToFormSubmit();
        this.subscribeToStatusChanges();
        this.subscribeToValueChanges();
    };
    /**
     * @return {?}
     */
    ValidationGroupDirective.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.subs.unsubscribe();
    };
    ValidationGroupDirective.decorators = [
        { type: Directive, args: [{
                    /* tslint:disable-next-line */
                    selector: '[formGroup],[formGroupName]',
                },] }
    ];
    /** @nocollapse */
    ValidationGroupDirective.ctorParameters = function () { return [
        { type: Injector },
        { type: ChangeDetectorRef },
        { type: FormGroupName, decorators: [{ type: Optional }, { type: Self }] },
        { type: FormGroupDirective, decorators: [{ type: Optional }, { type: Self }] },
        { type: ValidationGroupDirective, decorators: [{ type: Optional }, { type: SkipSelf }] }
    ]; };
    ValidationGroupDirective.propDecorators = {
        _blueprints: [{ type: Input, args: ['blueprints',] }],
        _errorTemplate: [{ type: Input, args: ['errorTemplate',] }],
        _invalidClasses: [{ type: Input, args: ['invalidClasses',] }],
        _mapErrorsFn: [{ type: Input, args: ['mapErrorsFn',] }],
        _skipValidation: [{ type: Input, args: ['skipValidation',] }],
        _targetSelector: [{ type: Input, args: ['targetSelector',] }],
        _validateOnSubmit: [{ type: Input, args: ['validateOnSubmit',] }]
    };
    return ValidationGroupDirective;
}(AbstractValidationDirective));
export { ValidationGroupDirective };
if (false) {
    /** @type {?} */
    ValidationGroupDirective.prototype.status$;
    /** @type {?} */
    ValidationGroupDirective.prototype.submit$;
    /** @type {?} */
    ValidationGroupDirective.prototype.value$;
    /** @type {?} */
    ValidationGroupDirective.prototype._blueprints;
    /** @type {?} */
    ValidationGroupDirective.prototype._errorTemplate;
    /** @type {?} */
    ValidationGroupDirective.prototype._invalidClasses;
    /** @type {?} */
    ValidationGroupDirective.prototype._mapErrorsFn;
    /** @type {?} */
    ValidationGroupDirective.prototype._skipValidation;
    /** @type {?} */
    ValidationGroupDirective.prototype._targetSelector;
    /** @type {?} */
    ValidationGroupDirective.prototype._validateOnSubmit;
    /**
     * @type {?}
     * @private
     */
    ValidationGroupDirective.prototype.subs;
    /** @type {?} */
    ValidationGroupDirective.prototype.injector;
    /** @type {?} */
    ValidationGroupDirective.prototype.cdRef;
    /** @type {?} */
    ValidationGroupDirective.prototype.groupName;
    /** @type {?} */
    ValidationGroupDirective.prototype.groupRef;
    /** @type {?} */
    ValidationGroupDirective.prototype.parentRef;
}
