// @abp/ng.core/esm2015/lib/services/environment.service.js
import { Injectable } from '@angular/core';
import { map } from 'rxjs/operators';
import { InternalStore } from '../utils/internal-store-utils';
import * as i0 from "@angular/core";
export class EnvironmentService {
    constructor() {
        this.store = new InternalStore({});
    }
    get createOnUpdateStream() {
        return this.store.sliceUpdate;
    }
    getEnvironment$() {
        return this.store.sliceState(state => state);
    }
    getEnvironment() {
        return this.store.state;
    }
    getApiUrl(key) {
        return (this.store.state.apis[key || 'default'] || this.store.state.apis.default).url;
    }
    getApiUrl$(key) {
        return this.store
            .sliceState(state => state.apis)
            .pipe(map((apis) => (apis[key || 'default'] || apis.default).url));
    }
    setState(environment) {
        this.store.set(environment);
    }
}
EnvironmentService.ɵprov = i0.ɵɵdefineInjectable({ factory: function EnvironmentService_Factory() { return new EnvironmentService(); }, token: EnvironmentService, providedIn: "root" });
EnvironmentService.decorators = [
    { type: Injectable, args: [{ providedIn: 'root' },] }
];
