// @abp/ng.core/esm2015/lib/services/multi-tenancy.service.js
import { Injectable } from '@angular/core';
import { RestService } from './rest.service';
import { SessionStateService } from './session-state.service';
import * as i0 from "@angular/core";
import * as i1 from "./rest.service";
import * as i2 from "./session-state.service";
export class MultiTenancyService {
    constructor(restService, sessionState) {
        this.restService = restService;
        this.sessionState = sessionState;
        this._domainTenant = null;
        this.isTenantBoxVisible = true;
        this.apiName = 'abp';
    }
    set domainTenant(value) {
        this._domainTenant = value;
        this.sessionState.setTenant(value);
    }
    get domainTenant() {
        return this._domainTenant;
    }
    /**
     * @deprecated Use AbpTenantService.findTenantByName method instead. To be deleted in v5.0.
     */
    findTenantByName(name, headers) {
        return this.restService.request({
            url: `/api/abp/multi-tenancy/tenants/by-name/${name}`,
            method: 'GET',
            headers,
        }, { apiName: this.apiName });
    }
    /**
     * @deprecated Use AbpTenantService.findTenantById method instead. To be deleted in v5.0.
     */
    findTenantById(id, headers) {
        return this.restService.request({ url: `/api/abp/multi-tenancy/tenants/by-id/${id}`, method: 'GET', headers }, { apiName: this.apiName });
    }
}
MultiTenancyService.ɵprov = i0.ɵɵdefineInjectable({ factory: function MultiTenancyService_Factory() { return new MultiTenancyService(i0.ɵɵinject(i1.RestService), i0.ɵɵinject(i2.SessionStateService)); }, token: MultiTenancyService, providedIn: "root" });
MultiTenancyService.decorators = [
    { type: Injectable, args: [{ providedIn: 'root' },] }
];
MultiTenancyService.ctorParameters = () => [
    { type: RestService },
    { type: SessionStateService }
];
